### PR TITLE
add test cases for FileSelector.tsx

### PR DESCRIPTION
**Pull request by namnguyen20999**
| Test Coverage Progress | File            | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s |
|------------------------|-----------------|---------|----------|---------|---------|-------------------|
| :hourglass: Before     | FileSelector.tsx | 64.28   | 45       | 30.76   | 63.76   | 37-39,71-85,96,102-104,110-111,116-124 |
| :white_check_mark: After  | FileSelector.tsx | 100     | 92.5     | 100     | 100     | 73,106-121       |

### CODE REVIEW

**Review by FabienLelaquais — DISMISSED** [review]
😯

File: frontend/taipy-gui/src/components/Taipy/FileSelector.tsx [hunk 1]
```diff
@@ -144,7 +144,7 @@ const FileSelector = (props: FileSelectorProps) => {
     }, [handleDrop, handleDragLeave, handleDragOverWithLabel]);
 
     return (
-        <label htmlFor={inputId} className={className}>
+        <label data-testid="file-selector" htmlFor={inputId} className={className}>
```
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/src/components/Taipy/FileSelector.tsx]: not sure I like this

File: frontend/taipy-gui/src/components/Taipy/FileSelector.tsx [hunk 2]
```diff
@@ -157,6 +157,7 @@ const FileSelector = (props: FileSelectorProps) => {
             <Tooltip title={hover || ""}>
                 <Button
                     id={id}
+                    data-testid="upload-button"
```
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/src/components/Taipy/FileSelector.tsx]: or this

File: frontend/taipy-gui/src/components/Taipy/FileSelector.spec.tsx [hunk 3]
```diff
@@ -98,4 +105,180 @@ describe("FileSelector Component", () => {
...
+        const elt = getByRole("button");
+
+        // Create a mock DragEvent
+        const mockEvent = new Event("dragleave", { bubbles: true }) as unknown as DragEvent;
+
+        // Dispatch the mock event
+        fireEvent(elt, mockEvent);
+
+        // Add assertions to check if dropLabel and dropSx have been reset
+        const dropLabelElement = getByTestId("file-selector");
```
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/src/components/Taipy/FileSelector.spec.tsx]: can't you get this element with the gettext(label) (default of specific) ?

File: frontend/taipy-gui/src/components/Taipy/FileSelector.spec.tsx [hunk 4]
```diff
@@ -98,4 +105,180 @@ describe("FileSelector Component", () => {
...
+        expect(dropLabelElement.textContent).toBe(" ");
+        const buttonElement = getByTestId("upload-button");
```
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/src/components/Taipy/FileSelector.spec.tsx]: is this different from  elt = getByRole("button"); ?

File: frontend/taipy-gui/src/components/Taipy/FileSelector.spec.tsx [hunk 5]
```diff
@@ -98,4 +105,180 @@ describe("FileSelector Component", () => {
...
+        const mockDispatch = jest.fn();
+        const { getByTestId } = render(
+            <TaipyContext.Provider value={{ state: INITIAL_STATE, dispatch: mockDispatch }}>
+                <FileSelector label="FileSelector" notify={true} />
+            </TaipyContext.Provider>,
+        );
+
+        // Simulate file upload
+        const file = new File(["(⌐□_□)"], "chucknorris.png", { type: "image/png" });
+        const inputElement = getByTestId("file-selector").querySelector("input");
```
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/src/components/Taipy/FileSelector.spec.tsx]: idem

File: frontend/taipy-gui/src/components/Taipy/FileSelector.spec.tsx [hunk 6]
```diff
@@ -98,4 +105,180 @@ describe("FileSelector Component", () => {
...
```
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/src/components/Taipy/FileSelector.spec.tsx]: idem

**Review by FredLL-Avaiga — DISMISSED** [review]
can you try and remove the getByTestId ?

File: frontend/taipy-gui/src/components/Taipy/FileSelector.tsx (hunk 1 above)
Comment (namnguyen20999) [reply frontend/taipy-gui/src/components/Taipy/FileSelector.tsx]: I've updated the test cases

File: frontend/taipy-gui/src/components/Taipy/FileSelector.tsx (hunk 2 above)
Comment (namnguyen20999) [reply frontend/taipy-gui/src/components/Taipy/FileSelector.tsx]: removed

**Review by FabienLelaquais — APPROVED** [review]
👌

File: frontend/taipy-gui/src/components/Taipy/FileSelector.spec.tsx (hunk 3 above)
Comment (namnguyen20999) [reply frontend/taipy-gui/src/components/Taipy/FileSelector.spec.tsx]: resolved

File: frontend/taipy-gui/src/components/Taipy/FileSelector.spec.tsx (hunk 4 above)
Comment (namnguyen20999) [reply frontend/taipy-gui/src/components/Taipy/FileSelector.spec.tsx]: resolved

**Pull request merged**